packages/template-retail-react-app/app/components/item-variant/item-attributes.jsx

```diff
@@ -38,25 +40,67 @@ const ItemAttributes = ({includeQuantity, currency, ...props}) => {
         }
     )
     const promos = res?.data || []
-    // Create a mapping of variation values to their associated attributes. This allows us
-    // the render the readable names/labels rather than variation value IDs.
-    const variationValues = Object.keys(variant.variationValues || []).map((key) => {
-        const value = variant.variationValues[key]
-        const attr = variant.variationAttributes?.find((attr) => attr.id === key)
-        return {
-            id: key,
-            name: attr?.name || key,
-            value: attr.values.find((val) => val.value === value)?.name || value
+    const variationValues = getDisplayVariationValues(
+        variant?.variationAttributes,
+        variant?.variationValues
+    )
+
+    // get variant info for bundled products in cart page and order history page
+    const productBundleIds =
+        variant?.bundledProductItems?.map(({productId}) => productId).join(',') ?? ''
+    const {data: productBundleVariantData, isLoading: bundleVariantIsLoading} = useProducts(
+        {
+            parameters: {
+                ids: productBundleIds,
+                allImages: false
+            }
+        },
+        {
+            enabled: Boolean(variant?.type?.bundle && productBundleIds),
+            select: (result) => {
+                return result?.data?.map((item) => {
```

Comment:
This is a bit tricky to parse at a glance - could you add a comment explaining the goal?